rpcs3/rpcs3qt/debuggerframe.h

```diff
@@ -67,35 +67,52 @@ class DebuggerFrame : public QDockWidget
 
 	u32 GetPc() const;
 	u32 CentrePc(u32 pc) const;
-	void OnSelectUnit();
 	//void resizeEvent(QResizeEvent* event);
 	void DoUpdate();
-	void ShowAddr(u32 addr);
 	void WriteRegs();
 
 	void OnUpdate();
+
+protected:
+	/** Override inherited method from Qt to allow signalling when close happened.*/
+	void closeEvent(QCloseEvent* event);
+
+signals:
+	void DebugFrameClosed();
+public slots:
+	void DoStep();
+private slots:
+	void OnSelectUnit();
 	void Show_Val();
 	void Show_PC();
 	void DoRun();
 	void DoPause();
-	void DoStep();
-	void keyPressEvent(QKeyEvent* event);
-	void mouseDoubleClickEvent(QMouseEvent* event);
+	void EnableUpdateTimer(bool state);
+};
 
-	void wheelEvent(QWheelEvent* event);
+class DebuggerList : public QListWidget
+{
+	Q_OBJECT
+
+	DebuggerFrame* m_debugFrame;
+
+public:
+	u32 m_pc;
+	u32 m_item_count;
+
+public:
+	DebuggerList(DebuggerFrame* parent);
+	void ShowAddr(u32 addr);
+
+private:
 	bool IsBreakPoint(u32 pc);
 	void AddBreakPoint(u32 pc);
 	void RemoveBreakPoint(u32 pc);
 
 protected:
-	/** Override inherited method from Qt to allow signalling when close happened.*/
-	void closeEvent(QCloseEvent* event);
```

Comment:
The close event was used for keeping the visibility menu in sync if the user hit the x button on the dock widget